pkg/net/rpc.go

```diff
@@ -91,6 +94,17 @@ func (o *Overlay) matchSample(s string) bool {
 	return o.matchJoin(string(base))
 }
 
+func (o *Overlay) matchGossip(s string) bool {
+	//	/casm/net/<version>/<ns>/gossip
+	base, subproto := protoutil.Split(protocol.ID(s))
+	if subproto != "gossip" {
+		return false
+	}
+
+	//	/casm/net/<version>/<ns>
+	return o.matchJoin(string(base))
+}
+
 /*
```

Comment:
Do we still need/want the random walk (and `Overlay.FindPeers`) now that we have this new gossip-based approach?

Reply:
Officially we don't, because the overlay can be maintained solely by the gossiping.

Reply:
Let's get rid of it, then.  The code was ugly, anyway, 😝